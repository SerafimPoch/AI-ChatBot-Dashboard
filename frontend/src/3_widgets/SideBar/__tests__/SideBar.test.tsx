import React from "react";
import { render } from "@testing-library/react";
import {
  SideBarContainer,
  SideBarLogoWrapper,
  SideBarListWrapper,
  SideBarLine,
} from "../SideBar.styles";

describe("SideBar page components", () => {
  it("SideBarContainer renders correctly", () => {
    const { asFragment } = render(<SideBarContainer expanded />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("SideBarLogoWrapper renders correctly", () => {
    const { asFragment } = render(<SideBarLogoWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("SideBarListWrapper renders correctly", () => {
    const { asFragment } = render(<SideBarListWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("SideBarLine renders correctly", () => {
    const { asFragment } = render(<SideBarLine />);
    expect(asFragment()).toMatchSnapshot();
  });
});
