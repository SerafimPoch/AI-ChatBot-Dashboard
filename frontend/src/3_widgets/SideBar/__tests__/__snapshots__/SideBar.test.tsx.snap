// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar page components SideBarContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 24px 20px;
  width: 260px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-radius: 20px;
  background: undefined;
}

@media (max-width:1550px) {
  .emotion-0 {
    padding: 24px 10px;
  }
}

@media (max-width:1000px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SideBar page components SideBarLine renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  height: 1px;
  background: #929292;
  margin: 8px 0;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SideBar page components SideBarListWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 8px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`SideBar page components SideBarLogoWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-bottom: 40px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
